### PR TITLE
Added Database Schema and bash files

### LINKED ISSUE #23: Create Database
### Steps
- [x] Define the database schema
- [x] Set up the database instance
- [x] Test the database setup to ensure it meets requirements

### User Story
As a data engineer, I need to create a database to store and manage the data required for the ETL and other processes.

### LINKED ISSUE #24: Develop Database Scripts
### Steps
- [x] Write scripts for database creation, management, and deletion (bash scripts - connect.sh, clear.sh)
- [x] Include scripts for tables, indexes, and other database objects
- [x] Validate the scripts for correctness and efficiency

### User Story
As a data engineer, I need scripts to create and manage the database to ensure consistent and efficient database operations.

### PR DESCRIPTION
**Pull request by danfh00**
Creates the database schema and bash files for connecting, clearing and querying the database.
Closes https://github.com/BerkayDur/sales-tracker-hermes/issues/23, https://github.com/BerkayDur/sales-tracker-hermes/issues/24

### CODE REVIEW

File: Database/schema.sql [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    user_id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    email TEXT NOT NULL,
+    CONSTRAINT user_email_unique UNIQUE (email)
+);
+
+CREATE TABLE subscriptions (
+    subscription_id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    user_id INTEGER NOT NULL REFERENCES users (user_id),
+    product_id INTEGER NOT NULL REFERENCES products (product_id),
+    price_threshold MONEY
```
Comment (NabihaMoh) [inline Database/schema.sql]: Is there a difference between INT and INTEGER? If not, maybe it's best to be consistent and use 1 throughout.

File: Database/schema.sql [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    url TEXT NOT NULL,
+    product_code TEXT NOT NULL,
+    product_name TEXT NOT NULL,
+    CONSTRAINT product_url_unique UNIQUE (url)
+);
+
+CREATE TABLE users (
+    user_id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    email TEXT NOT NULL,
+    CONSTRAINT user_email_unique UNIQUE (email)
```
Comment (NabihaMoh) [inline Database/schema.sql]: Is this the same as having UNIQUE at the end of the variable?

File: Database/schema.sql [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    subscription_id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    user_id INTEGER NOT NULL REFERENCES users (user_id),
+    product_id INTEGER NOT NULL REFERENCES products (product_id),
+    price_threshold MONEY
+);
+
+CREATE TABLE price_readings (
+    reading_id INT GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+    product_id INTEGER NOT NULL REFERENCES products (product_id),
+    timestamp TIMESTAMP(0) NOT NULL,
```
Comment (NabihaMoh) [inline Database/schema.sql]: Maybe 'reading_at' is a more descriptive name than timestamp?

**Review by NabihaMoh — CHANGES_REQUESTED** [review]
Some comments to have a look at before merging.

File: Database/schema.sql (hunk 1 above)
Comment (danfh00) [reply Database/schema.sql]: Thanks, now changed all to INTEGER

File: Database/schema.sql (hunk 2 above)
Comment (danfh00) [reply Database/schema.sql]: Yes, thanks this is now moved to the variable

File: Database/schema.sql (hunk 3 above)
Comment (danfh00) [reply Database/schema.sql]: Changed to reading_at

**Review by NabihaMoh — APPROVED** [review]

**Pull request merged**